src/lib/common/statistics.cpp

```diff
@@ -232,170 +234,52 @@ std::string timingStatistics(std::string indent, Format format, std::string apiV
   bool lastReqTime              = (lastTimeStat.reqTime.tv_sec != 0)              || (lastTimeStat.reqTime.tv_nsec != 0);
   bool lastXmlParseTime         = (lastTimeStat.xmlParseTime.tv_sec != 0)         || (lastTimeStat.xmlParseTime.tv_nsec != 0);
 
-  bool  accXmlParseTimeComma         = false;
-  bool  accReqTimeComma              = accXmlParseTime           || accXmlParseTimeComma;
-  bool  accRenderTimeComma           = accReqTime                || accReqTimeComma;
-  bool  accMongoBackendTimeComma     = accRenderTime             || accRenderTimeComma;
-  bool  accMongoReadWaitTimeComma    = accMongoBackendTime       || accMongoBackendTimeComma;
-  bool  accMongoWriteWaitTimeComma   = accMongoReadWaitTime      || accMongoReadWaitTimeComma;
-  bool  accMongoCommandWaitTimeComma = accMongoWriteWaitTime     || accMongoWriteWaitTimeComma;
-  bool  accJsonV2ParseTimeComma      = accMongoCommandWaitTime   || accMongoCommandWaitTimeComma;
-  bool  accJsonV1ParseTimeComma      = accJsonV2ParseTime        || accJsonV2ParseTimeComma;
-
-  bool  lastXmlParseTimeComma         = false;
-  bool  lastReqTimeComma              = lastXmlParseTime         || lastXmlParseTimeComma;
-  bool  lastRenderTimeComma           = lastReqTime              || lastReqTimeComma;
-  bool  lastMongoBackendTimeComma     = lastRenderTime           || lastRenderTimeComma;
-  bool  lastMongoReadWaitTimeComma    = lastMongoBackendTime     || lastMongoBackendTimeComma; 
-  bool  lastMongoWriteWaitTimeComma   = lastMongoReadWaitTime    || lastMongoReadWaitTimeComma;
-  bool  lastMongoCommandWaitTimeComma = lastMongoWriteWaitTime   || lastMongoWriteWaitTimeComma;
-  bool  lastJsonV2ParseTimeComma      = lastMongoCommandWaitTime || lastMongoCommandWaitTimeComma;
-  bool  lastJsonV1ParseTimeComma      = lastJsonV2ParseTime      || lastJsonV2ParseTimeComma;
-
   bool last = lastJsonV1ParseTime || lastJsonV2ParseTime || lastMongoBackendTime || lastRenderTime || lastReqTime || lastXmlParseTime;
   bool acc  = accJsonV1ParseTime || accJsonV2ParseTime || accMongoBackendTime || accRenderTime || accReqTime || accXmlParseTime;
 
-  std::string  out     = "";
-  std::string  indent2 = indent;
-
   if (!acc && !last)
   {
     timeStatSemGive(__FUNCTION__, "no stats to report");
-    return "";
+    return "{}";
   }
 
-  if (format == JSON)
+  JsonHelper jh;
+
+  if (acc)
+  {
+    JsonHelper accJh;
+
+    if (accJsonV1ParseTime)      accJh.addFloat("jsonV1Parse",      timeSpecToFloat(accTimeStat.jsonV1ParseTime));
```

Comment:
I think this (and a similar case for last) is a justified exception to the style rule of using `{` and `}`, in order to make the code more clean and readable.